### PR TITLE
Few changes per our initial discussion

### PR DESCRIPTION
**Pull request by stkonst**
As we discussed via e-mail

### CODE REVIEW

File: draft-spaghetti-grow-nrtm-v4.xml [hunk 1]
```diff
@@ -173,13 +185,14 @@
       </t>
       <t>
         In other words, either the mirror server guarantees that clients following the deltas have a correct and complete view, or MUST reinitialise, which will force clients to reinitialise as well.
+        !<-- SK: Should we define a procedure for that??? -->
```
Comment (mxsasha) [inline draft-spaghetti-grow-nrtm-v4.xml:188]: A procedure for reinitialising? It's the same procedure as for the first initialisation. The session ID rotation will force clients to reload.

File: draft-spaghetti-grow-nrtm-v4.xml [hunk 2]
```diff
@@ -376,8 +389,8 @@
 
     <section title="Purpose">
       <t>
-        The Update Notification File is used by mirror clients to discover whether any changes exist between the state of the IRR mirror server and the mirror client's state.
-        It also describes the location of the files containing the snapshot and incremental deltas.
+        The Update Notification File is generated by the mirror server and used by mirror clients to discover whether any changes exist between the state of the IRR mirror server and the mirror client's state.
+        It also describes the location of the files containing the Snapshot File and incremental deltas.
```
Comment (mxsasha) [inline draft-spaghetti-grow-nrtm-v4.xml:393]:
```suggestion
        It also describes the location of the Snapshot File and incremental Delta Files.
```

File: draft-spaghetti-grow-nrtm-v4.xml [hunk 3]
```diff
@@ -570,9 +584,10 @@
     <section title="Purpose">
       <t>
         A Delta File contains all changes for exactly one incremental update of the IRR Database.
-        It may include new, modified and deleted objects. Delta Files can contain multiple updates to multiple objects.
+        It may include new, modified and deleted objects. Delta Files can contain multiple alternations to multiple objects.
```
Comment (mxsasha) [inline draft-spaghetti-grow-nrtm-v4.xml:587]:
```suggestion
        It may include new, modified and deleted objects. Delta Files can contain multiple alterations to multiple objects.
```

File: draft-spaghetti-grow-nrtm-v4.xml [hunk 4]
```diff
@@ -570,9 +584,10 @@
...
       </t>
     </section>
+    <!--SK: Under which circumstances a new Delta file is published? Is this something that we need to define? -->
```
Comment (mxsasha) [inline draft-spaghetti-grow-nrtm-v4.xml:590]: Does https://htmlpreview.github.io/?https://github.com/mxsasha/nrtmv4/blob/main/draft-spaghetti-grow-nrtm-v4.html#name-delta-files-2 not suffice for that? About once a minute, if there were changes, with catching up permitted after e.g. downtime.

**Comment by mxsasha** [comment]
Good improvements! I left a few comments in the PR on your inline comments.

File: draft-spaghetti-grow-nrtm-v4.xml (hunk 1 above)
Comment (stkonst) [reply draft-spaghetti-grow-nrtm-v4.xml:188]: Hi Sasha... I think it was not clear on my side so let me rephrase it. What I mean is that in the draft is stated that the server is responsible to ensure that the clients have the correct & complete view of deltas. Should it be defined a process of detecting faults and a method to overcome those or we should leave it up to the server developers and focus only on the transport part?

File: draft-spaghetti-grow-nrtm-v4.xml (hunk 4 above)
Comment (stkonst) [reply draft-spaghetti-grow-nrtm-v4.xml:590]: Hi Sasha, yes it is quite sufficient. But upon reviewing the document I goth the feeling that is not a fixed "per-minute" timeframe but can be "up-to-a-minute", which means we allow some room for earlier/faster delta files. Is that correct or I misunderstood it? Also in the same first bullet of 3.3.1. says " if there have been changes to RPSL objects in that time frame". Question is what if there are no changes? We skip the publication of Delta files or we publish a delta file with zero changes, regardless ? If we skip the publication, then the next one is published in the +2 or +3 minute ? 

I am not sure If I described my question correctly, if not I am happy to elaborate again.

File: draft-spaghetti-grow-nrtm-v4.xml (hunk 1 above)
Comment (mxsasha) [reply draft-spaghetti-grow-nrtm-v4.xml:188]: I think we focus more here on the transport part.

The scenario I had in mind is that there are some situations in IRRD where admins can discard the journal for a source. In that case, IRRD can't generate a contiguous set of deltas anymore for all changes that happened, therefore must reinitialise, forcing clients to reinitialise. Basically, a server either knows that it has complete history that it writes/wrote to delta files, or it might have missed things, and must reinitialise.

File: draft-spaghetti-grow-nrtm-v4.xml (hunk 4 above)
Comment (mxsasha) [reply draft-spaghetti-grow-nrtm-v4.xml:590]: I see. What I had in mind is this: every minute, a mirror server looks up what was included in the last delta file it published. If there have been changes to RPSL objects between "now" and that last file, produce a delta file.

This means:
* No changes, no delta file. We would just clutter the repo.
* The one minute publication timeframe does not have to be exact to the second. But not 5 minutes either. The idea is that this keeps the mirroring delay to one minute on average.
* If somehow the delta producer did not run, e.g. because that process was shut down, it will produce one bigger delta file upon the next run (if there were changes in the mean time).

Regarding timing, I'm thinking you might have:
```
10:15:10: Delta publisher runs, finds no changes, stops
10:15:11: An RPSL object is changed
10:16:10: Delta publisher runs, finds one change and publishes it, stops
10:17:10: Delta publisher runs, finds no changes, stops
10:18:10: Delta publisher runs, finds no changes, stops
10:18:53: An RPSL object is changed
10:19:10: Delta publisher runs, finds two changes and publishes them, stops
10:20:10: Delta publisher runs, finds no changes, stops
```

**Comment by mxsasha** [comment]
I'm going to go ahead and merge this without the inline comments so we can keep moving, but do make issues or more PRs if my answers haven't addressed those enough :)
